pkg/trustedcabundle/trustedcabundle.go

```diff
@@ -0,0 +1,153 @@
+package trustedcabundle
+
+import (
+	"context"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dsci "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+)
+
+const (
+	injectionOfCABundleAnnotatoion = "security.opendatahub.io/inject-trusted-ca-bundle"
+	CAConfigMapName                = "odh-trusted-ca-bundle"
+	CADataFieldName                = "odh-ca-bundle.crt"
+)
+
+func IsValidNamespace(ns client.Object) bool {
+	if !strings.HasPrefix(ns.GetName(), "openshift-") && !strings.HasPrefix(ns.GetName(), "kube-") &&
+		ns.GetName() != "default" && ns.GetName() != "openshift" && !HasCABundleAnnotationDisabled(ns) {
+		return true
+	}
+	return false
+}
+
+func HasCABundleAnnotationDisabled(ns client.Object) bool {
+	nsAnnotations := ns.GetAnnotations()
+
+	for annKey, annValue := range nsAnnotations {
+		if annKey == injectionOfCABundleAnnotatoion && annValue == "false" {
+			return true
+		}
+	}
+	return false
+}
+
+func AddCABundleConfigMapInAllNamespaces(ctx context.Context, cli client.Client, dscInit *dsci.DSCInitialization) error {
+	namespaceList := &corev1.NamespaceList{}
+	err := cli.List(ctx, namespaceList)
+	if err != nil {
+		return err
+	}
+
+	for _, ns := range namespaceList.Items {
+		if !strings.HasPrefix(ns.Name, "openshift-") && !strings.HasPrefix(ns.Name, "kube-") &&
+			ns.Name != "default" && ns.Name != "openshift" {
```

Comment:
one last piece, from https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/concepts/projects/utils.ts 
"system" is opt-out, should we exclude it here too?

Reply:
Not adding `system` to the excluded namespaces, as it is not included in ocp default namespaces.